oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoUtils.java

```diff
@@ -190,4 +194,15 @@ static Type getDocumentStoreExceptionTypeFor(Throwable t) {
         }
         return type;
     }
+
+    /**
+     * Util method to get node size limit for current mongo version
+     *
+     * @param version version of current mongo db
+     * @return size limit based on mongo db version
+     */
+    static int getNodeNameLimit(final String version) {
+        final ServerVersion sv = new ServerVersion(Arrays.stream(version.split("\\.")).map(Integer::new).collect(Collectors.toList()));
+        return sv.compareTo(new ServerVersion(4,2)) >= 0 ? Integer.MAX_VALUE : Utils.NODE_NAME_LIMIT;
```

Comment:
This can be further simplified. I noticed MongoDB versions can also have non-digit characters. Though, these versions should be limited to release candidates.

https://www.mongodb.com/docs/manual/reference/versioning/#release-candidate--rc--releases

I think it would still be nice if Oak can also run on a release candidate.

The lines could be replaced with a call to MongoStatus.isVersion(4, 2).
https://github.com/apache/jackrabbit-oak/blob/c5eb75abce6cac9fd9e9244d0fd34b1345a285e8/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoStatus.java#L160

I'll create a PR for your feature branch.

Reply:
@rishabhdaim I noticed you updated this PR with changes I proposed. However, some changes are missing. The tests you added to MongoUtilsTest and I updated are now missing in this PR. Can you please add them again?

Reply:
@mreutegg I have removed those tests because I added your proposed changes to use `MongoStatus.isVersion` in the `MongoDocumentStore` class.

Reply:
I see. On the other hand there are no more tests that check if the name length limit is handled correctly for different MongoDB versions.

Reply:
I got it. I have reverted my change and merged your PR in my local fork.